error.go

```diff
@@ -34,11 +35,44 @@ import (
 // Errors which know their underlying cause should implement this interface to
 // be compatible with RootCause.
 //
-// We use an unexported "cause" method instead of "Cause" because we don't
-// want dig-internal causes to be confused with the cause of the user-provided
-// errors. (For example, if the users are using github.com/pkg/errors.)
+// We use unexported methods because we don't want dig-internal causes to be
+// confused with the cause of the user-provided errors. For example, if we
+// used Unwrap(), then user-provided methods would also be unwrapped by
+// RootCause. We want RootCause to eliminate the dig error chain only.
 type causer interface {
+	fmt.Formatter
+
+	// Returns the next error in the chain.
```

Comment:
s/next/preceding/g ?

Reply:
No, next. If you visualize the causal relationship between errors as a linked list, the cause of an error is the next node in the linked list.

Reply:
Just to make sure I have this straight: In the context of 
```
cannot provide function "go.uber.org/dig".TestProvideConstructorErrors\S+ \
```
the cause would be
```
"cannot specify a name for result objects: dig.out embeds dig.Out"
```
As described in ln 1297 of dig_test.go

Reply:
Yeah, in "cannot do x: reason y,"  if the current error is "cannot do x", "reason y" is next/cause.

Reply:
👍